File: packages/kotlin/src/main/kotlin/com/openbancor/catalogmx/cfdi/CfdiResources.kt
package com.openbancor.catalogmx.cfdi

import com.openbancor.catalogmx.utils.SharedData

object CfdiResources {
    fun satUrlToLocalPath(url: String): String {
        val uri = java.net.URI(url)
        return SharedData.resolvePath("sat/xsd/resources/${uri.host}${uri.path}").absolutePath
    }

    fun cfdi40Xsd(): String = satUrlToLocalPath(
        "http://www.sat.gob.mx/sitio_internet/cfd/4/cfdv40.xsd"
    )

    fun cadenaOriginal40Xslt(): String = satUrlToLocalPath(
        "http://www.sat.gob.mx/sitio_internet/cfd/4/cadenaoriginal_4_0/cadenaoriginal_4_0.xslt"
    )
}

File: packages/dart/lib/src/cfdi/cfdi_resources.dart
/// CFDI resources helper (paths for XSD/XSLT)
library;

class CfdiResourceLocator {
  final String basePath;

  const CfdiResourceLocator({required this.basePath});

  String satUrlToLocalPath(String url) {
    final uri = Uri.parse(url);
    return '$basePath/sat/xsd/resources/${uri.host}${uri.path}';
  }

  String cfdi40Xsd() {
    return satUrlToLocalPath(
        'http://www.sat.gob.mx/sitio_internet/cfd/4/cfdv40.xsd');
  }

  String cadenaOriginal40Xslt() {
    return satUrlToLocalPath(
      'http://www.sat.gob.mx/sitio_internet/cfd/4/cadenaoriginal_4_0/cadenaoriginal_4_0.xslt',
    );
  }
}
